openpype/plugins/publish/collect_hierarchy.py

```diff
@@ -30,14 +30,15 @@ def process(self, context):
 
             # shot data dict
             shot_data = {}
-            family = instance.data.get("family")
+            family = instance.data["family"]
+            families = instance.data["families"]
 
             # filter out all unepropriate instances
             if not instance.data["publish"]:
                 continue
 
             # exclude other families then self.families with intersection
-            if not set(self.families).intersection([family]):
+            if not set(self.families).intersection(set(families + [family])):
```

Comment:
@tokejepsen this is just an improvement of the code. The test was only performed on main family and not on additional families.